inferno-core/src/Inferno/Parse.hs

```diff
@@ -731,12 +730,13 @@ mkOperators opsTable =
 parseExpr ::
   OpsTable ->
   Map.Map ModuleName OpsTable ->
+  [CustomType] ->
   Text ->
   Either
     (NonEmpty (ParseError Text InfernoParsingError, SourcePos))
     (Expr () SourcePos, [Comment SourcePos])
-parseExpr opsTable modOpsTables s =
-  case runParser (runWriterT $ flip runReaderT (opsTable, modOpsTables) $ topLevel expr) "<stdin>" s of
+parseExpr opsTable modOpsTables customTypes s =
```

Comment:
This function is an entrypoint to the parser, so `customTypes` is an argument to it